metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -241,650 +281,569 @@ class MetalsLanguageServer(
       parseTrees ::
       compilations.pauseables
   )
+
   private val timerProvider: TimerProvider = new TimerProvider(time)
   private val trees = new Trees(buffers, scalaVersionSelector)
-  private val documentSymbolProvider = new DocumentSymbolProvider(trees)
+
+  private val documentSymbolProvider = new DocumentSymbolProvider(
+    trees,
+    Some(initializeParams).supportsHierarchicalDocumentSymbols,
```

Comment:
Extension method can operate on plain `InitializeParams` rather than Option of it

Reply:
Are you sure? I didn't see extensions methods in `MetalsEnrichments` for this. Or do you mean we should switch that?

Reply:
It is `XtensionClientCapabilities` which operates on `initializeParams: Option[l.InitializeParams]`

Reply:
Yes, exactly, that's what I mean, it operates on an `Option`. Does your original comment mean you'd like to change this to operate directly on `l.InitializeParams` instead of the `Option`?